src/pages/index.js

```diff
@@ -32,23 +42,120 @@ addPostButton.addEventListener("click", () => {
   postFormPopup.open();
 });
 
+//Handle Submit functions
+function handleSubmit(request, popupModal, popupForm, renderingText, postText) {
+  renderLoading(true, popupForm, renderingText, postText);
+  request()
+    .then(() => {
+      popupModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      renderLoading(false, popupForm, renderingText, postText);
+    });
+}
+
+const handleProfilePictureSubmit = ({ picture }) => {
+  function makeRequest() {
+    return api.updateAvatar(picture).then((profileInfo) => {
+      profilePicture.src = profileInfo.avatar;
+    });
+  }
+
+  handleSubmit(makeRequest, profilePictureFormPopup, editProfilePictureForm);
+};
+
 const handleProfileFormSubmit = ({ name, job }) => {
-  userInfo.setUserInfo({ name, job });
-  profileFormPopup.close();
+  function makeRequest() {
+    return api.updateProfileInfo({ name, about: job }).then((newData) => {
+      userInfo.setUserInfo({
+        name: newData.name,
+        job: newData.about,
+      });
+    });
+  }
+
+  handleSubmit(makeRequest, profileFormPopup, editProfileForm);
 };
 
 const handlePostFormSubmit = ({ title, url }) => {
   const newPost = {
     name: title,
     link: url,
   };
-  const newCard = createCard(newPost);
-  cardSection.addNewItem(newCard);
-  postFormPopup.close();
+
+  function makeRequest() {
+    return api.createApiCard(newPost).then((createdCard) => {
+      cardSection.addNewItem(createCard(createdCard));
+    });
+  }
+
+  handleSubmit(makeRequest, postFormPopup, addPostForm);
 
   formValidators[selectors.addPostForm].disableButton();
```

Comment:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.